pyaedt/modeler/stackup_3d.py

```diff
@@ -0,0 +1,2284 @@
+
+    def __init__(self, application):
+        self._app = application
+        self._layer_name = []
+        self._layer_position = []
+        self._dielectric_list = []
+        self._dielectric_name_list = []
+        self._ground_list = []
+        self._ground_name_list = []
+        self._ground_fill_material = []
+        self._signal_list = []
+        self._signal_name_list = []
+        self._signal_material = []
+        self._object_list = []
+        self._vias = []
+        self._end_of_stackup3D = NamedVariable(self._app, "StackUp_End", "0mm")
+        self._z_position_offset = 0
+        self._first_layer_position = "layer_1_position"
+        self._shifted_index = 0
+        self._stackup = OrderedDict({})
+        self._start_position = NamedVariable(self._app, self._first_layer_position, "0mm")
+        self._dielectric_x_position = NamedVariable(self._app, "dielectric_x_position", "0mm")
+        self._dielectric_y_position = NamedVariable(self._app, "dielectric_y_position", "0mm")
+        self._dielectric_width = NamedVariable(self._app, "dielectric_width", "1000mm")
+        self._dielectric_length = NamedVariable(self._app, "dielectric_length", "1000mm")
+        self._padstacks = []
+
+    @property
+    def padstacks(self):
+        """List of padstack created.
+
+        Returns
+        -------
+        List
+        """
+        return self._padstacks
+
+    @property
+    def dielectrics(self):
+        """List of dielectric created.
+
+        Returns
+        -------
+        List
+        """
+        return self._dielectric_list
+
+    @property
+    def grounds(self):
+        """List of ground created.
+
+        Returns
+        -------
+        List
+        """
+        return self._ground_list
+
+    @property
+    def signals(self):
+        """List of signals created.
+
+        Returns
+        -------
+        List
+        """
+        return self._signal_list
+
+    @property
+    def objects(self):
+        """List of obects created.
+
+        Returns
+        -------
+        List
+        """
+        return self._object_list
+
+    @property
+    def objects_by_layer(self):
+        """List of padstack created.
+
+        Returns
+        -------
+        List
+        """
+        objs = {}
+        for obj in self.objects:
+            if objs.get(obj.layer_name, None):
+                objs[obj.layer_name].append(obj)
+            else:
+                objs[obj.layer_name] = [obj]
+        return objs
+
+    @property
+    def start_position(self):
+        """Variable containing the start position.
+
+        Returns
+        -------
+        :class:`pyaedt.modeler.stackup_3d.NamedVariable`
+        """
+        return self._start_position
+
+    @start_position.setter
+    def start_position(self, expression):
+        self._start_position.expression = expression
+
+    @property
+    def dielectric_x_position(self):
+        """Stackup Starting origin x variable.
```

Comment:
This description should be rewritten--It consists of 4 adjectives in front of the noun "variable". I truly don't understand the description.